extensions/in-memory/negotiation-store-memory/src/main/java/org/eclipse/dataspaceconnector/negotiation/store/memory/CriterionToPredicateConverter.java

```diff
@@ -0,0 +1,56 @@
+/*
+ *  Copyright (c) 2020 - 2022 Microsoft Corporation
+ *
+ *  This program and the accompanying materials are made available under the
+ *  terms of the Apache License, Version 2.0 which is available at
+ *  https://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  SPDX-License-Identifier: Apache-2.0
+ *
+ *  Contributors:
+ *       Microsoft Corporation - initial API and implementation
+ *
+ */
+
+package org.eclipse.dataspaceconnector.negotiation.store.memory;
+
+import org.eclipse.dataspaceconnector.spi.query.Criterion;
+import org.eclipse.dataspaceconnector.spi.query.CriterionConverter;
+import org.eclipse.dataspaceconnector.spi.types.domain.contract.negotiation.ContractNegotiation;
+
+import java.util.Arrays;
+import java.util.Objects;
+import java.util.function.Predicate;
+
+import static org.eclipse.dataspaceconnector.negotiation.store.memory.ContractNegotiationFunctions.property;
+
+/**
+ * Converts a {@link Criterion} into a {@link Predicate} for use with an {@link InMemoryContractNegotiationStore}.
+ * Currently, only the "=" and "in" operators are supported (cf. {@link Criterion#getOperator()}).
+ */
+class CriterionToPredicateConverter implements CriterionConverter<Predicate<ContractNegotiation>> {
```

Comment:
There's another `CriterionToPredicateConverter`, and these classes are pretty much the same. Eventually the `property` method that I commented above could be part of a `ReflectionUtils` or something like that and give the property for any kind of object it receives.

Reply:
I'll create a `ReflectionUtils` class, we'll probably need this in several other places as well.
Then, a base `CriterionToPredicateConverter` will become easy.